src/storage-controller/src/lib.rs

```diff
@@ -935,6 +935,81 @@ where
         Ok(())
     }
 
+    async fn alter_ingestion_export_data_configs(
+        &mut self,
+        // Map from primary ingestion id to source-export id and new data config
+        source_exports: BTreeMap<GlobalId, (GlobalId, SourceExportDataConfig)>,
```

Comment:
What happens here when there are multiple exports? It looks like it will lead to some ingestion commands being sent where some exports point to the old connection data and some to the new when the alter should appear as a single atomic change

Reply:
The `ingestion_id` here corresponds to the primary source id, such that multiple exports for a given source being updated will just trigger one `run_ingestion` call for the primary 'ingestion'/source id. Is that what you are referring to?

Reply:
Ah you are totally right!! This map can only have one value per ingestion id - I will fix!